uwblueprint/focus-on-nature
# Nathan tina/add discount amount to confirmation email

## Notion ticket link

[Add discount amount to payment confirmation page & confirmation email](https://www.notion.so/uwblueprintexecs/Add-discount-amount-to-payment-confirmation-page-confirmation-email-cf67b336a1e840fab829548f115a4b05?pvs=4)



## Implementation description
* Added retrieveStripeCheckoutSession to stripe.utils to grab a checkout session by the chargeId
* Included discount amount when sending confirmation email to parents



## Steps to test
** Note: you must be running the local Stripe CLI in order to test this change. (brew install stripe/stripe-cli/stripe; stripe login; stripe listen -forward-to localhost:500/campers/confirm-payment)
1. Go through the register flow to signup for a camp
2. Once redirected to the Stripe checkout page use code _TEST15_  to apply 15% discount
3. You'll be redirected back to FON 
4. The camper will be updated in the database to hasPaid = true
5. Verify in the confirmation email that discount amount applied is shown (example below)
<img width="414" alt="Screen Shot 2023-07-15 at 4 47 04 PM" src="https://github.com/uwblueprint/focus-on-nature/assets/88808428/663774cb-a7eb-4164-8d52-e92cf405937b">


## What should reviewers focus on?
* After being redirected to FON, we receive an error modal saying the payment was cancelled. This bug should be handled in a separate ticket.  


## Checklist
